plugin/src/main/java/org/opensearch/ml/model/MLModelManager.java

```diff
@@ -2077,40 +2077,43 @@ public void getConnector(String connectorId, String tenantId, ActionListener<Con
             .tenantId(tenantId)
             .build();
 
-        sdkClient.getDataObjectAsync(getDataObjectRequest).whenComplete((r, throwable) -> {
-            log.debug("Completed Get Connector Request, id:{}", connectorId);
-            if (throwable != null) {
-                Exception cause = SdkClientUtils.unwrapAndConvertToException(throwable);
-                if (ExceptionsHelper.unwrap(cause, IndexNotFoundException.class) != null) {
-                    log.error("Failed to get connector index", cause);
-                    listener.onFailure(new OpenSearchStatusException("Failed to find connector", RestStatus.NOT_FOUND));
+        try (ThreadContext.StoredContext ctx = client.threadPool().getThreadContext().stashContext()) {
+            sdkClient.getDataObjectAsync(getDataObjectRequest).whenComplete((r, throwable) -> {
+                log.debug("Completed Get Connector Request, id:{}", connectorId);
```

Comment:
We can restore the context here? `ctx.restore();`?

Reply:
wdyt should I add that in? It doesn't look like the context is referenced inside this block at all but it doesn't hurt to restore the context once the action has finished either.

Reply:
FYI I will be driving a campaign shortly to remove and replace usages of stashContext: https://github.com/opensearch-project/opensearch-plugins/issues/238

Example of replacing the calls in geospatial plugin: https://github.com/opensearch-project/geospatial/pull/715

Reply:
>wdyt should I add that in? It doesn't look like the context is referenced inside this block at all but it doesn't hurt to restore the context once the action has finished either.

The "action" is to get the object from the system index which should be done when we reach at line 2082. That's why I was suggesting to restore the context something like [this](https://github.com/opensearch-project/flow-framework/blob/6d6116d1bb688787f06a58fc0f6c2d9b09854007/src/main/java/org/opensearch/flowframework/transport/DeleteWorkflowTransportAction.java#L149)

Reply:
Added a line to restore the context once the action has completed.